eng/dashboardpack/dashboardpack.csproj

```diff
@@ -26,8 +26,10 @@
 
   <Target Name="Build" />
 
-  <Target Name="BeforeBuild" BeforeTargets="Build">
-    <MSBuild Projects="../../src/Aspire.Dashboard/Aspire.Dashboard.csproj" Targets="publish" Properties="Configuration=$(Configuration);Platform=$(Platform);TargetFramework=$(TargetFramework);RuntimeIdentifier=$(DashboardRuntime)" />
+  <Target Name="BeforeBuild" BeforeTargets="_GetPackageFiles">
+    <!-- separate restore step to allow targets from nugets to get imported in the publish step -->
+    <MSBuild Projects="../../src/Aspire.Dashboard/Aspire.Dashboard.csproj" Targets="restore" Properties="Configuration=$(Configuration);Platform=$(Platform);TargetFramework=$(TargetFramework);RuntimeIdentifier=$(DashboardRuntime)" />
+    <MSBuild Projects="../../src/Aspire.Dashboard/Aspire.Dashboard.csproj" Targets="publish" Properties="Configuration=$(Configuration);Platform=$(Platform);TargetFramework=$(TargetFramework);RuntimeIdentifier=$(DashboardRuntime);_Some_unique_prop=$([System.Guid]::NewGuid())" />
```

Comment:
Can you explain why are we passing in this extra property? This is breaking incremental build. Essentially Restore is happening again given there are new global properties.

Reply:
Is this solving a particular issue for the helix test scenario? Aspire.Dashboard should already be restored as part of the main build before this project runs.

Reply:
This was a stale change which I had to add earlier because a restored nuget package was adding targets, but the project was not re-evaluated thus the targets were not available.
Subsequently, I had changed the way we install the workload for testing, but missed removing this change. Thank you for questioning it!

Reply:
This wasn't stale, instead the conditions were reproducible only on CI. 
TL;dr - this was needed because it needed fresh restore, and a project re-evaluation to load targets from `grpc.tools` nuget.

Details:

On CI we used `./build.sh` for the main build, and then `.dotnet.sh build` to install the workload, which hit the target above.
- `build.sh` - sets `NUGET_PACKAGES` to `$repoRoot/.packages`
- and `./dotnet.sh build` defaulted to `$HOME/.nuget/packages`

So,
1. `./build.sh` - restores nugets to `$repoRoot/.packages`, and imports targets from the grpc.tools nuget.
2. `.dotnet build` - runs the above line and runs `publish` target
    a. which imports the `nuget.g.{props,targets}` for the E2E project, but it fails to import the `grpc.tools` targets because it is now looking at the wrong path to the restored package!
    b. Even if the `restore` target is run before this `publish`, the msbuild project is not re-evaluated and thus the `gprc.tools` targets don't get imported
Instead, if we `restore`, and then `publish` but with a new property added then it triggers a re-evaluation of the project, which causes the `grpc.tools` targets to get imported, and run

Now that I have figured out the reason for all this mess - `./build.sh vs ./dotnet.sh build`, I have:
1. changed to use `./build.sh` for both the cases
2. And have the target instead be triggered based on a property passed to the command on CI only.

This fixes the issue correctly.